numpy/__init__.pyi

```diff
@@ -1016,6 +1016,7 @@ def sort(
     axis: Optional[int] = ...,
     kind: Optional[_SortKind] = ...,
     order: Union[None, str, Sequence[str]] = ...,
+    by: Union[None, Sequence[ndarray]] = ...,
```

Comment:
Does this specifically have to be an `ndarray` or will any array-like object suffice?

Reply:
yeah any array like should suffice, will update this part

Reply:
Will the `ndarray` be interpreted as a single key or a tuple of keys? I think the former is desired.

Reply:
@hameerabbasi  isn't it a tuple of keys we support in lexsort?

Reply:
Concretely, are we doing `tuple(by)` to get the desired keys (which may result in `len(by)` keys if `by` is an `ndarray`), or are we wrapping it into a tuple by doing `by=(by,)`?

Reply:
Correct me if i am wrong, but to be consistent with np.lexsort, doing tuple(by) which will result in len(keys)  is what seems to be happening here. (lexsort accepts k (N,) shaped tuple elements or (k, N) array). I would also prefer `by=(by,)` if there arent other places in numpy which have similar behavior like lexsort.